spec/forms/training_details_form_spec.rb

```diff
@@ -48,18 +48,32 @@
         end
       end
 
-      context "duplicate" do
+      context "duplicate active trainee" do
         let!(:existing_trainee) { create(:trainee, trainee_id: "Test123") }
-        let(:trainee) { build(:trainee, provider: existing_trainee.provider, trainee_id: "TEST123") }
+        let(:trainee) { build(:trainee, provider: existing_trainee.provider, trainee_id: existing_trainee.trainee_id) }
 
         it "returns a duplicate error message" do
+          expect(existing_trainee.inactive?).to be(false)
           expect(subject.errors[:trainee_id]).to include(
             I18n.t("#{error_attr}.trainee_id.uniqueness"),
           )
           expect(subject.duplicate_error?).to be_truthy
         end
       end
 
+      context "duplicate inactive trainee" do
+        let!(:existing_trainee) { create(:trainee, trainee_id: "Test123", state: "withdrawn") }
+        let(:trainee) { build(:trainee, provider: existing_trainee.provider, trainee_id: existing_trainee.trainee_id) }
+
+        it "returns a duplicate error message" do
+          expect(existing_trainee.inactive?).to be(true)
```

Comment:
Again, you can extract this line into a separate example:

```suggestion
it "existing trainee becomes inactive" do
  expect(existing_trainee.inactive?).to be(true)
end 

```